### ISSUE TITLE
url parameter "x-client-Ver" break the behaviour of https://login.microsoftonline.com (i.e. reset password feature)

### ISSUE THREAD

**Issue by alrouen** [AzureAD/azure-activedirectory-library-for-js#188]
Hi,

In a SPA using Adal.js, when the user is redirected to https://login.microsoftonline.com, several url parameters are added to this login page (tenant ID, nonce, redirect url, etc.).

One of them is "x-client-Ver", with as value is the Adal.js lib version number (i.e.: 1.0.7).

When the user is on the login page and use the "Can’t access your account?" link (https://passwordreset.microsoftonline.com), the arrive on a empty and meaningless Microsoft page just saying:

    Can't access your account?
    Visit passwordreset.microsoftonline.com from a browser to reset your password.

Instead we should get the regular reset password form of Azure Active Director:

    Who are you?
    To recover your account, begin by entering your user ID and the characters in the picture or audio below.
    
    User ID : <input with user id coming from previous page>
    
    <captcha>
    
    [Next Button] [Cancel Button]

To get this expected reset page, I just need to remove "x-client-Ver" parameter from the https://login.microsoftonline.com page, then reload, then test the "Can’t access your account?" link.

From what I found in the Adal.js library, this parameter is added by this function:

```
AuthenticationContext.prototype._addClientId = function () {
    // x-client-SKU
    // x-client-Ver
    return '&x-client-SKU=Js&x-client-Ver=' + this._libVersion();
};
```
https://github.com/AzureAD/azure-activedirectory-library-for-js/blob/master/lib/adal.js#L1188-L1192

If I remove the "x-client-Ver" from what this function returns, I'm getting the expected behaviour.

In the meantime, I suppose this parameter to be used for reporting purpose in AAD.

In conclusion either this parameter is setting an unexpected value that break the behaviour of https://login.microsoftonline.com, or there is a bug in the reset password page with this "x-client-ver" parameter.

Could you confirm and propose a workaround ?

Thanks!

**Comment by tushargupta51** [comment]
This is an issue with how AAD interprets this paramter when resetting password. We have given feedback to them. We need to send this parameter for reporting purposes. Closing this issue as it wont be fixed in ADAL but in AAD.
